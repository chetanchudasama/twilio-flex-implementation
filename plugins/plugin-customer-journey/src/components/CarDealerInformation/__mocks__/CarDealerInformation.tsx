import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
