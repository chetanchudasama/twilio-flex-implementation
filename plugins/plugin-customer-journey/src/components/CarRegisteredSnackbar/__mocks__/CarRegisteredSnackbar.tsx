import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
