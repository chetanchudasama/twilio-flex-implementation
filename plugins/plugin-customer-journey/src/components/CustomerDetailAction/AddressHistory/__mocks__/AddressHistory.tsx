import React from "react";
import { AddressHistoryProps } from "../AddressHistory";

const AddressHistory: React.FC<AddressHistoryProps> = (props) => (
  <div className="address-history" />
);

export default AddressHistory;
