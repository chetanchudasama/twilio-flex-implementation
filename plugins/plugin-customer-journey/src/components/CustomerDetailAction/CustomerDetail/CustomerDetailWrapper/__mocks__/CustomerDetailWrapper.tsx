import React from "react";
import { ComponentProps } from "../CustomerDetailWrapper";

const CustomerDetailWrapper: React.FC<ComponentProps> = (props) => (
  <div className="customer-detail-wrapper" />
);

export default CustomerDetailWrapper;
