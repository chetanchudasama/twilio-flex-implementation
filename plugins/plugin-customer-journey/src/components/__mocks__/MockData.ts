import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
