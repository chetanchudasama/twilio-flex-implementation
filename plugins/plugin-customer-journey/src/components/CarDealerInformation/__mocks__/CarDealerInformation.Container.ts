import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
