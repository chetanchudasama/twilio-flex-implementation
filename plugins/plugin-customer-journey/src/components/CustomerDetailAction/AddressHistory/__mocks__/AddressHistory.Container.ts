import AddressHistory from "./AddressHistory";

export default AddressHistory;
