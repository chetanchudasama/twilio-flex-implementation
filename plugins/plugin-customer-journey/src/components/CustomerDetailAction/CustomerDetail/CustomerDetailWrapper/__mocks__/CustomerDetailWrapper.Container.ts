import CustomerDetailWrapper from "./CustomerDetailWrapper";

export default CustomerDetailWrapper;
