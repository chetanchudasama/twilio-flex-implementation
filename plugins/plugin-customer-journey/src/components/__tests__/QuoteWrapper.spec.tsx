import React from "react";
import { create } from "react-test-renderer";
import flushPromises from "flush-promises";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { Manager } from "@twilio/flex-ui";
import {
  QuoteWrapper,
  QuoteWrapperProps,
} from "../Wizard/QuotesStep/QuoteWrapper/QuoteWrapper";
import { QuoteDetailModel } from "../../models/QuoteDetailModel";
import { flexManagerMock } from "../__mocks__/MockData";

jest.mock("../../services/application.service");
jest.mock("../Wizard/QuotesStep/SavedQuoteList/SavedQuoteList.Container");
jest.mock("../Wizard/QuotesStep/SavedQuoteList/SavedQuoteList");

const mockGetQuoteList = jest.fn();

jest.mock("../../services/application.service", () => ({
  useApplicationService: jest.fn(() => ({
    getQuoteList: mockGetQuoteList,
  })),
}));

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    WizardStepWrapper: jest.fn(({ children, onNext, onPrevious }) => {
      return (
        <div>
          <button type="button" className="onNext" onClick={onNext}>
            onNext
          </button>
          <button type="button" className="onPrevious" onClick={onPrevious}>
            onPrevious
          </button>
          {children}
        </div>
      );
    }),
  };
});

const defaultProps: QuoteWrapperProps = {
  applicationId: 0,
  setSavedQuotes: jest.fn(() => {}),
  moveForward: jest.fn(() => {}),
  moveBackward: jest.fn(() => {}),
};

const props: QuoteWrapperProps = defaultProps;

Manager.getInstance = flexManagerMock;

describe("QuoteWrapper", () => {
  beforeEach(() => {
    mockGetQuoteList.mockResolvedValueOnce(() =>
      Promise.resolve(new QuoteDetailModel())
    );
  });

  afterEach(() => {
    mockGetQuoteList.mockClear();

    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it("renders correctly, without any props", async () => {
    const wrapper = create(<QuoteWrapper {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, with props", async () => {
    props.moveForward = jest.fn();
    props.moveBackward = jest.fn();
    const wrapper = create(<QuoteWrapper {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("when next button clicked and move forward event is provided, move forward event should be emitted", async () => {
    const moveForward = jest.fn();
    props.moveForward = moveForward;
    const wrapper = mount(<QuoteWrapper {...props} />);
    await flushPromises();
    wrapper.find(".onNext").last().simulate("click");

    expect(moveForward).toHaveBeenCalledTimes(1);
  });

  it("when previous button clicked and move backward event is provided, move backward event should be emitted", async () => {
    const moveBackward = jest.fn();
    props.moveBackward = moveBackward;
    const wrapper = mount(<QuoteWrapper {...props} />);
    await flushPromises();
    wrapper.find(".onPrevious").last().simulate("click");

    expect(moveBackward).toHaveBeenCalledTimes(1);
  });

  it("should fetch quote list when application id is present", async () => {
    props.applicationId = 1;

    mockGetQuoteList.mockImplementationOnce(() =>
      Promise.resolve(new QuoteDetailModel())
    );

    await act(async () => {
      const wrapper = mount(<QuoteWrapper {...props} />);
      wrapper.update();
      await flushPromises();
      wrapper.update();
    });

    expect(mockGetQuoteList).toHaveBeenCalledTimes(1);
  });

  it("should not fetch quote data if application id is not present", async () => {
    props.applicationId = 0;

    mockGetQuoteList.mockImplementationOnce(() =>
      Promise.resolve(new QuoteDetailModel())
    );

    await act(async () => {
      const wrapper = mount(<QuoteWrapper {...props} />);
      wrapper.update();
      await flushPromises();
      wrapper.update();
    });

    expect(mockGetQuoteList).not.toHaveBeenCalled();
  });
});
