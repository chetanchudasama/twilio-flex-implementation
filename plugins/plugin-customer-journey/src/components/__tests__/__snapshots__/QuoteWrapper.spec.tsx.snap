// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteWrapper renders correctly, with props 1`] = `
<div
  className="css-8ton2x"
>
  <div>
    <button
      className="onNext"
      onClick={[Function]}
      type="button"
    >
      onNext
    </button>
    <button
      className="onPrevious"
      onClick={[Function]}
      type="button"
    >
      onPrevious
    </button>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`QuoteWrapper renders correctly, without any props 1`] = `
<div
  className="css-8ton2x"
>
  <div>
    <button
      className="onNext"
      onClick={[Function]}
      type="button"
    >
      onNext
    </button>
    <button
      className="onPrevious"
      onClick={[Function]}
      type="button"
    >
      onPrevious
    </button>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
